src/System.Runtime.Numerics/tests/BigInteger/TryWriteBytes.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+    public partial class ExtractBytesMembersTests
+    {
+        [Theory]
+        [MemberData(nameof(FromIntTests_MemberData))]
+        public void TryWriteBytes_FromIntTests(int i, byte[] expectedBytes) =>
+            ValidateGetByteCountAndTryWriteBytes(new BigInteger(i), expectedBytes);
+
+        [Theory]
+        [MemberData(nameof(FromLongTests_MemberData))]
+        public void TryWriteBytes_FromLongTests(long l, byte[] expectedBytes) =>
+            ValidateGetByteCountAndTryWriteBytes(new BigInteger(l), expectedBytes);
+
+        [Theory]
+        public void TryWriteBytes_FromStringTests(string str, byte[] expectedBigEndianBytes)
+        {
+            byte[] expectedBytes = (byte[])expectedBigEndianBytes.Clone();
+            Array.Reverse(expectedBytes);
+            ValidateGetByteCountAndTryWriteBytes(BigInteger.Parse(str), expectedBytes);
+        }
+
+        private void ValidateGetByteCountAndTryWriteBytes(BigInteger bi, byte[] expectedBytes)
+        {
+            byte[] bytes = bi.ToByteArray();
+            Assert.Equal(expectedBytes, bytes);
+
+            int count = bi.GetByteCount();
+            Assert.Equal(expectedBytes.Length, count);
+
+            int bytesWritten;
+            for (int i = 0; i < 2; i++)
```

Comment:
I'm sure it's a matter of taste, but it feels odd to me to use a loop just to repeat the same logic twice with a different array length. It gives me the impression that the for loop is significant in some way, but it's just a clever trick to re-use logic.